test/e2e/resize_nodes.go

```diff
@@ -230,109 +270,213 @@ var _ = Describe("ResizeNodes", func() {
 		if err := c.Namespaces().Delete(ns); err != nil {
 			Failf("Couldn't delete namespace '%s', %v", ns, err)
 		}
-		By("restoring the original node instance group size")
-		if err := resizeNodeInstanceGroup(testContext.CloudConfig.NumNodes); err != nil {
-			Failf("Couldn't restore the original node instance group size: %v", err)
-		}
-		if err := waitForNodeInstanceGroupSize(testContext.CloudConfig.NumNodes); err != nil {
-			Failf("Couldn't restore the original node instance group size: %v", err)
-		}
-		if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes); err != nil {
-			Failf("Couldn't restore the original cluster size: %v", err)
-		}
 	})
 
-	testName = "should be able to delete nodes."
-	It(testName, func() {
-		Logf("starting test %s", testName)
-
-		if !providerIs(supportedProviders...) {
-			By(fmt.Sprintf("Skipping %s test, which is only supported for providers %v (not %s)",
-				testName, supportedProviders, testContext.Provider))
-			return
-		}
-
-		if testContext.CloudConfig.NumNodes < 2 {
-			By(fmt.Sprintf("skipping %s test, which requires at lease 2 nodes (not %d)",
-				testName, testContext.CloudConfig.NumNodes))
-			return
-		}
-
-		// Create a replication controller for a service that serves its hostname.
-		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
-		name := "my-hostname-delete-node-" + string(util.NewUUID())
-		replicas := testContext.CloudConfig.NumNodes
-		createServeHostnameReplicationController(c, ns, name, replicas)
-		pods, err := waitForPodsCreated(c, ns, name, replicas)
-		Expect(err).NotTo(HaveOccurred())
-		e := waitForPodsRunning(c, pods)
-		if len(e) > 0 {
-			Failf("Failed to wait for pods running: %v", e)
-		}
-		err = verifyPodsResponding(c, ns, name, pods)
-		Expect(err).NotTo(HaveOccurred())
-
-		By(fmt.Sprintf("decreasing cluster size to %d", replicas-1))
-		err = resizeNodeInstanceGroup(replicas - 1)
-		Expect(err).NotTo(HaveOccurred())
-		err = waitForNodeInstanceGroupSize(replicas - 1)
-		Expect(err).NotTo(HaveOccurred())
-		err = waitForClusterSize(c, replicas-1)
-		Expect(err).NotTo(HaveOccurred())
-
-		By("verifying whether the pods from the removed node are recreated")
-		pods, err = waitForPodsCreated(c, ns, name, replicas)
-		Expect(err).NotTo(HaveOccurred())
-		e = waitForPodsRunning(c, pods)
-		if len(e) > 0 {
-			Failf("Failed to wait for pods running: %v", e)
-		}
-		err = verifyPodsResponding(c, ns, name, pods)
-		Expect(err).NotTo(HaveOccurred())
+	Describe("Resize", func() {
+		AfterEach(func() {
+			if !providerIs(supportedProviders...) {
+				return
```

Comment:
You'll need to notify the user that the test has been skipped here.  Ideally we would tell the Ginkgo test framework that the test has been skipped, but I don't think that's currently possible.  So I'd suggest failing (Failf(...)) with a message along the lines of:

"Resize test not supported on provider %s.   You can avoid this failure by using ginkgo.skip=Resize in your environment."